packages/dev/core/src/Materials/Textures/equiRectangularCubeTexture.ts

```diff
@@ -152,20 +153,25 @@ export class EquiRectangularCubeTexture extends BaseTexture {
         if (!scene) {
             return;
         }
-        this._texture = scene
+        const faceDataArrays = callback();
+
+        const texture = scene
             .getEngine()
-            .createRawCubeTextureFromUrl(
-                this.url,
-                scene,
+            .createRawCubeTexture(
+                faceDataArrays,
                 this._size,
                 Constants.TEXTUREFORMAT_RGB,
                 scene.getEngine().getCaps().textureFloat ? Constants.TEXTURETYPE_FLOAT : Constants.TEXTURETYPE_UNSIGNED_INTEGER,
                 this._noMipmap,
-                callback,
-                null,
-                this._onLoad,
-                this._onError
+                false,
+                Constants.TEXTURE_TRILINEAR_SAMPLINGMODE

```

Comment:
Note here uses default values from `createRawCubeTextureFromUrl`, but since there is a `samplingMode` in class `ThinTexture`, should this also be configurable?

Reply:
No that's ok (and that's what the previous code was doing anyway), the user can change the sampling mode afterwards by calling `updateSamplingMode` if they want.